Ruby Path - Ruby on Rails -> Controlers: Swaps whitelist for allowlist (#24008)
TheOdinProject/curriculum: ruby_on_rails/rails_basics/controllers.md

The important distinction between the "scalar" parameter values like strings and more complex parameters like hashes and arrays is that Rails 4 implemented some protections in the controller, called "Strong Parameters".  This is so the user can't send you harmful data (like automatically setting themselves as an admin user when they create an account).  To do this, Rails makes you explicitly verify that you are willing to accept certain items of a hash or array.

> *Note: This used to be done in Rails 3 by setting `attr_accessible` in the model to allowlist attributes, so you will probably see that in a lot of Stack Overflow posts and earlier applications.*

Reviewer: ```suggestion
*Note: This used to be done in Rails 3 by setting `attr_accessible` in the model to allow attributes, so you will probably see that in a lot of Stack Overflow posts and earlier applications.*
```

> To "allowlist", or explicitly allow, parameters, you use the methods `require` and `permit`.  Basically, you `require` the name of your array or hash to be in Params (otherwise it'll throw an error), and then you `permit` the individual attributes inside that hash to be used.  For example:

Reviewer: ```suggestion
To explicitly allow parameters, you use the methods `require` and `permit`.  Basically, you `require` the name of your array or hash to be in Params (otherwise it'll throw an error), and then you `permit` the individual attributes inside that hash to be used.  For example:
```

~~~ruby
  def allowed_post_params
    params.require(:post).permit(:title,:body,:author_id)
  end
~~~

> This will allowlist and return the hash of only those params that you specified (e.g. `{:title => "your title", :body => "your body", :author_id => "1"}` ).  If you didn't do this, when you tried to access params[:post] nothing would show up! Also, if there were any additional fields submitted inside the hash, these will be stripped away and made inaccessible (to protect you).

Reviewer: Apologies @MLGabe, just one more.
```suggestion
This will return the hash of only those params that you explicitly allow (e.g. `{:title => "your title", :body => "your body", :author_id => "1"}` ).  If you didn't do this, when you tried to access params[:post] nothing would show up! Also, if there were any additional fields submitted inside the hash, these will be stripped away and made inaccessible (to protect you).
```

Author: No problem, sorry for not picking it up!
Already changed :)
